### PR TITLE
[spi_device] Functional coverage flash and pass

### PR DESCRIPTION
**Pull request by kosta-kojdic**
Functional coverage proposal for flash and passthrough modes.
Added merged tests to hjson file to be mapped.

Signed-off-by: Kosta Kojdic <kosta.kojdic@ensilica.com>

### CODE REVIEW

File: hw/ip/spi_device/data/spi_device_testplan.hjson [hunk 1]
```diff
@@ -445,5 +445,81 @@
...
+    {
+      name: spi_device_buffer_boundary_cg
+      desc: '''
+            Cover buffer boundary crossing (buffer flip).'''
+    }
+    {
+      name: spi_device_lanes_cg
+      desc: '''
+            Cover out commands that use dual mode.
+            Cover out comands that use quad mode.'''
```
Comment (weicaiyang) [inline hw/ip/spi_device/data/spi_device_testplan.hjson:522]: nit: comands -> commands

Comment (weicaiyang) [reply hw/ip/spi_device/data/spi_device_testplan.hjson:522]: cross this with flash and passthrough mode

File: hw/ip/spi_device/data/spi_device_testplan.hjson [hunk 2]
```diff
@@ -445,5 +445,81 @@
...
+            Cover payload size of upload transaction exceeds 256B limit (wrap around).'''
+    }
+    {
+      name: spi_device_read_commands_cg
+      desc: '''
+            Cover read status commands.
+            Cover read JEDEC command.
+            Cover read SFDP command.
+            Cover all other read commands configurable in slots up to 10.
+            Cover dummy cycle for all read commands.'''
```
Comment (weicaiyang) [inline hw/ip/spi_device/data/spi_device_testplan.hjson:498]: cover `INTERCEPT_EN` as well.

**Review by weicaiyang — COMMENTED** [review]
@eunchan please also take a look

**Review by eunchan — COMMENTED** [review]
LGTM

**Review by weicaiyang — APPROVED** [review]
LGTM

**Pull request merged**